Dockerfile

```diff
@@ -1,49 +1,65 @@
-# Build the manager binary
-FROM golang:1.13 as builder
-
-WORKDIR /workspace
-# Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN go mod download
-
-# Copy the go source
-COPY main.go main.go
-COPY api/ api/
-COPY controllers/ controllers/
-COPY pkg/ pkg/
-COPY tools/ tools/
-
-# Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o csv-generator tools/csv-generator.go
-
-FROM registry.access.redhat.com/ubi7/ubi-minimal:latest
+# golang-builder is used in OSBS build
+ARG GOLANG_BUILDER=golang:1.13
+ARG OPERATR_BASE_IMAGE=registry.access.redhat.com/ubi7/ubi-minimal:latest
+
+FROM ${GOLANG_BUILDER} AS builder
+
+# Intended to build in OSBS using cachito external sources bundle
+ARG REMOTE_SOURCE=.
+ARG REMOTE_SOURCE_DIR=keystone-operator
+ARG REMOTE_SOURCE_SUBDIR=.
+ARG DEST_ROOT=/dest-root
+ARG GO_BUILD_EXTRA_ARGS="-v"
+ARG GO_BUILD_EXTRA_ENV="CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on"
+
+COPY $REMOTE_SOURCE $REMOTE_SOURCE_DIR
+WORKDIR ${REMOTE_SOURCE_DIR}/${REMOTE_SOURCE_SUBDIR}
+
+RUN mkdir -p ${DEST_ROOT}/usr/local/bin/
+
+RUN ${GO_BUILD_EXTRA_ENV} go build ${GO_BUILD_EXTRA_ARGS} -a -o ${DEST_ROOT}/usr/local/bin/manager main.go
+RUN ${GO_BUILD_EXTRA_ENV} go build ${GO_BUILD_EXTRA_ARGS} -a -o ${DEST_ROOT}/usr/local/bin/csv-generator tools/csv-generator.go
+
+RUN cp tools/user_setup ${DEST_ROOT}/usr/local/bin/
+RUN cp -r templates ${DEST_ROOT}/templates
+
+# prep the bundle
+RUN mkdir -p ${DEST_ROOT}/bundle
+RUN cp config/crd/bases/keystone.openstack.org_keystoneapis.yaml ${DEST_ROOT}/bundle/keystone.openstack.org_keystoneapis_crd.yaml
+RUN cp config/crd/bases/keystone.openstack.org_keystoneendpoints.yaml ${DEST_ROOT}/bundle/keystone.openstack.org_keystoneendpoints_crd.yaml
+RUN cp config/crd/bases/keystone.openstack.org_keystoneservices.yaml ${DEST_ROOT}/bundle/keystone.openstack.org_keystoneservices_crd.yaml
+
+# strip top 2 lines (this resolves parsing in opm which handles this badly)
+RUN sed -i -e 1,2d ${DEST_ROOT}/bundle/*
+
+FROM ${OPERATR_BASE_IMAGE}
+ARG DEST_ROOT=/dest-root
+
+LABEL   com.redhat.component="keystone-operator-container" \
```

Comment:
I wonder if those can be args as well? I mean like all per operator keep as vars, this way any new operator will have exactly same bottom part of Dockerfile and only vary with beginning ?

Reply:
could, but let's get incrementally closer between upstream and downstream and then look at across the different operators

If I recall correctly there are currently 3 or 4 variants